text/0039-service-overhaul.md

```diff
@@ -0,0 +1,196 @@
+***This RFC is dependent on [RFC 38](0038-protocol-api.md).***
+
+# Summary
+
+- Make OBS able to accept third-party service plugins
+- Attribute a id for each services rather than a common one
+- A service can be provided with multiple protocols
+- Separate Twitch, Restream and YouTube integration and make them plugins
+
+# Motivation
+
+Actually even if OBS has the service API, developer can't create third-party service plugin because there is no mechanism to list those and show their own property view.
+
+Before in OBS, in the Stream settings page, there was a combobox with two choice:
+- Streaming Services
+- Custom Streaming Server
+
+Each one had his own property view because those two are the two actual registered official services.
+This behavior could enable third-party service plugin to exist.
+
+Nowadays in OBS, this page show the list of services with many new elements showed of hidden depending of the selection (like recommended settings). With also Twitch and Restream OAuth integrations. And no use of the property views provided by `rtmp-services`.
+
+We need to restore the usage of property views and the possibility to create and use service plugin.
+
+It will also provide the ability for some stream services to be able to made their own plugin.
+
+*Since using the service API was not possible with the actual OBS, any breaking change in it (if there are) will not impact any plugin for OBS.*
+
+# Design
+
+## UI
+The Stream settings page will only contain a combo box listing registered services with a remade "Show All" option. So property view can be shown under it.
+The "short list" with "Show All" option, will be harcoded with a list of service id in the page code.
+Any removed element will be reimplemented in a new way.
+
+(Same for auto-wizard)
```

Comment:
Please elaborate here. Does the wizard need new APIs too?  eg. bandwidth testing, stream testing, etc.

Reply:
Feel free to open a new conversation about this if needed.